dmpr.py

```diff
@@ -456,224 +432,213 @@ def _recalculate_routing_table(self):
+                            path_found = True
+                            break
+                    if path_found == False:
+                        neigh_data['paths']["{}>{}".
+                            format(self._conf["id"], neigh_name)].append(interface_name)
+                    found_neigh = True
+                    break
+            if found_neigh == False:
+                neigh_routing_paths = self._add_neigh_entries(interface_name, sender_name,
+                                                              sender_data, neigh_routing_paths)
         else:
-             neigh_routing_paths = self._add_neigh_entries(interface_name, sender_name, 
-                                                           sender_data, neigh_routing_paths)
+            neigh_routing_paths = self._add_neigh_entries(interface_name, sender_name,
+                                                          sender_data, neigh_routing_paths)
         return neigh_routing_paths
 
-
     def _add_neigh_entries(self, interface_name, sender_name, sender_data, neigh_routing_paths):
         neigh_routing_paths['neighs'][sender_name] = {'next-hop': sender_name,
                                                       'networks': sender_data['msg']['networks'],
-                                                      'paths':{"{}>{}".
-                                                               format(self._conf["id"],sender_name):
-                                                               [interface_name]}
-                                                     }
+                                                      'paths': {"{}>{}".
+                                                                    format(self._conf["id"], sender_name):
+                                                                    [interface_name]}
+                                                      }
         return neigh_routing_paths
 
-
     def _add_all_othernodes_bw(self, sender_name, sender_data, neigh_routing_paths):
         othernode_bw = dict()
         othernode_bw = neigh_routing_paths['othernode_paths']['high_bandwidth'].copy()
         if not sender_name in othernode_bw:
-           othernode_bw[sender_name] = dict()
-           othernode_bw[sender_name]['path_characteristics'] = dict()
+            othernode_bw[sender_name] = dict()
+            othernode_bw[sender_name]['path_characteristics'] = dict()
         else:
-             self.log.info('updating high bandwidth other nodes from the existing neighbour')
+            self.log.info('updating high bandwidth other nodes from the existing neighbour')
         othernode_bw[sender_name] = sender_data['msg']['routingpaths']['high_bandwidth'].copy()
-        othernode_bw[sender_name]['path_characteristics'] = sender_data['msg']['routingpaths']['path_characteristics'].copy()
+        othernode_bw[sender_name]['path_characteristics'] = sender_data['msg']['routingpaths'][
+            'path_characteristics'].copy()
```

Comment:
not sure if copy is really required or if the whole datastructure can be better coded ... Please check

Reply:
will be obsolete soon